src/emulate.c

```diff
@@ -374,7 +457,7 @@ enum {
 #if RV32_HAS(GDBSTUB)
 #define RVOP_NO_NEXT(ir) (!ir->next | rv->debug_mode)
 #else
-#define RVOP_NO_NEXT(ir) (!ir->next)
+#define RVOP_NO_NEXT(ir) (!ir->next | rv->is_interrupted)
```

Comment:
Shall you update the case for `RV32_HAS(GDBSTUB)`?

Reply:
> Shall you update the case for `RV32_HAS(GDBSTUB)`?

In that case, we have to introduce a new variable to indicate the state whether in block emulation or exception/interrupt.

Reply:
How about calling `is_trapped`?

Reply:
> How about calling `is_trapped`?

It sounds good.